### ISSUE TITLE
Date is not displayed in the menu bar when Hidden Bar is collapsed

### ISSUE THREAD

**Issue by igRo** [dwarvesf/hidden#207]
### Describe the bug

Date is not displayed in the menu bar when Hidden Bar is collapsed.

### Steps to reproduce

1. Click on Hidden Bar icon in the Menu Bar to collapse it

### Expected behaviour

Date/time is not affected and displayed (f.e. "Tue 17. May 21:35:10")

### Actual behaviour

Date/time is corrupted, "Tue 21:35:10" is displayed

**Screenshots**

<img width="592" alt="Screenshot 2022-05-17 at 21 34 41" src="https://user-images.githubusercontent.com/376776/168895855-4a58e50b-2373-4afb-90b3-46c2e1b2372a.png">

<img width="362" alt="Screenshot 2022-05-17 at 21 34 48" src="https://user-images.githubusercontent.com/376776/168895885-15bb8d4a-1788-4bd1-9a89-a02b112a53ef.png">

### Environment

 - macOS Monterey 12.4
 - Hidden Bar 1.8 (12) from MAS

**Comment by lifanh** [comment]
I have the same problem here.
In addition, when Hidden Bar is collapsed and I click on the date and time part, the highlight part goes to the wrong place.
<img width="748" alt="image" src="https://user-images.githubusercontent.com/19503212/168943767-bdd611fa-45cf-4758-97b4-8c08000fab15.png">
If Hidden Bar is expanded and everything seems to be ok.
<img width="888" alt="image" src="https://user-images.githubusercontent.com/19503212/168943748-168dbd70-9a75-4d9d-b896-f7aca2b60f93.png">

**Comment by shangdongyang** [comment]
Same bugs above, hope developers can pay attention to them.

**Comment by int0x19** [comment]
Change settings to always show date
<img width="760" alt="Снимок экрана 2022-05-18 в 10 53 54" src="https://user-images.githubusercontent.com/3350877/168967080-367e12a5-fa34-4e5c-b23b-028ba9d077cc.png">

**Comment by shangdongyang** [comment]
> Change settings to always show date <img alt="Снимок экрана 2022-05-18 в 10 53 54" width="760" src="https://user-images.githubusercontent.com/3350877/168967080-367e12a5-fa34-4e5c-b23b-028ba9d077cc.png">

This setting can display the date. Thank you. But I still hope developers can fix it, ha ha

**Comment by lifanh** [comment]
> Change settings to always show date <img alt="Снимок экрана 2022-05-18 в 10 53 54" width="760" src="https://user-images.githubusercontent.com/3350877/168967080-367e12a5-fa34-4e5c-b23b-028ba9d077cc.png">

It works! Thank you!

**Comment by conny3496** [comment]
I guess this was introduced on purpose so it does not need fixing. There should be a toggle though to influence this as it pleases.

**Comment by ReeganExE** [comment]
Seems like we all get this issue after upgrading to **macOS Monterey 12.4**.
Temporary fix by changing the Dock & Menu settings as int0x19 suggested:
![image](https://user-images.githubusercontent.com/7277418/169010497-9fc74473-dd3e-48ff-aa2a-1c77b7da5b5e.png)

**Comment by igRo** [comment]
> Change settings to always show date

Thanks! Can confirm, that solves the problem for me.

I guess, the issue can be closed now.

**Comment by DimitriDR** [comment]
> > Change settings to always show date
> 
> Thanks! Can confirm, that solves the problem for me.
> 
> I guess, the issue can be closed now.

I don't agree, we shouldn't close the issue as the given answers are workarounds, not a bug fix.

**Comment by mayamcdougall** [comment]
Maybe include a note somewhere around the clock portion of the `Hidden | Shown` diagram?

It seems like the macOS behavior itself is very much working as intended.  The "Available Space" shrunk, so macOS hid the date.  Making a workaround for that "feature" sounds counter-intuitive... but so is the default user experience with this setting enabled.

So, maybe just a note to the user in case they want to change it.

But that's just my two cents... and I only discovered this app about 10 minutes ago.  Then immediately had this issue. 😂

(The app seems great so far though. 😉
Much better option than the Freemium app I was using.
I discovered it via `alternativeto.net` in case you're curious.)

**Comment by dipakparmar** [comment]
spent 30 minutes figuring out why the date is missing and killed all the possible services to figure out if that was just random or a bug. and later, I'm here on this issue... I think having an option in the app itself will help this fixing this tiny issue. 

In my case, `always show` does not work it still hides.

**Comment by dploeger** [comment]
@dipakparmar In my case I just had to restart Hidden Bar to make it work after setting it in the settings.

**Comment by dipakparmar** [comment]
@dploeger I got it working after reinstalling it.

**Comment by DamianPereira** [comment]
Even if not showing the date is considered ok, the wrong highlight part of this should still be considered a bug. Anyway, hidden menubar should not "reduce" available space when collapsing.

**Comment by kainchow** [comment]
来个中文的
<img width="668" alt="333" src="https://user-images.githubusercontent.com/28430952/183793989-3eec7bc4-f78c-4f8f-82b9-2aa6d4642d3e.png">

**Comment by chusiang** [comment]
Thanks all! <[_ _]>

I fixed this problem, and I recorded it at https://github.com/dwarvesf/hidden/issues/218#issuecomment-1215015175 .

Enjoy it.

**Comment by kud** [comment]
![screenshot-2022-10-27-11-04-22@2x](https://user-images.githubusercontent.com/655838/198255990-79807382-a9c5-4f14-a43b-5ea0a466268d.png)
<img width="193" alt="screenshot-2022-10-27-11-04-28" src="https://user-images.githubusercontent.com/655838/198256024-d52f87f6-75aa-487a-89f5-01237f8d0522.png">

It did not work for me. :/

**Comment by kud** [comment]
Ha sorry! Clicking twice on "Show the day of the week" made the trick. Thanks.
